applications/RomApplication/python_scripts/rom_analysis.py

```diff
@@ -83,14 +110,26 @@ def ModifyAfterSolverInitialize(self):
             nodal_dofs = len(self.project_parameters["solver_settings"]["rom_settings"]["nodal_unknowns"].GetStringArray())
             rom_dofs = self.project_parameters["solver_settings"]["rom_settings"]["number_of_rom_dofs"].GetInt()
 
-            # Set the nodal ROM basis
+            # Set the right nodal ROM basis 
             aux = KratosMultiphysics.Matrix(nodal_dofs, rom_dofs)
             for node in computing_model_part.Nodes:
                 node_id = str(node.Id)
                 for j in range(nodal_dofs):
                     for i in range(rom_dofs):
                         aux[j,i] = nodal_modes[node_id][j][i].GetDouble()
                 node.SetValue(KratosROM.ROM_BASIS, aux)
+            
+            # Set the left nodal ROM basis if it is different than the right nodal ROM basis (i.e. Petrov-Galerkin)
+            if (self.solving_strategy == "Petrov-Galerkin"):
+                petrov_galerkin_nodal_modes = self.rom_parameters["petrov_galerkin_nodal_modes"]
+                petrov_galerkin_nodal_dofs = len(self.project_parameters["solver_settings"]["rom_settings"]["nodal_unknowns"].GetStringArray())
+                aux = KratosMultiphysics.Matrix(petrov_galerkin_nodal_dofs, self.petrov_galerkin_rom_dofs)
+                for node in computing_model_part.Nodes:
+                    node_id = str(node.Id)
```

Comment:
I guess this is required because our key is a string value. If so, maybe we can consider using an integer as key to avoid the string comparison, which is more expensive. OFC, this is out of the scope of this PR.